readline-lib/readline/rep.rkt

```diff
@@ -5,11 +5,14 @@
 
 (define-runtime-path rep-start "rep-start.rkt")
 
-(provide install-readline!)
+(provide install-readline!
+         pre-readline-input-port)
 
-(let ([inp (current-input-port)] [outp (current-output-port)])
-  (when (and (eq? 'stdin (object-name inp)) (terminal-port? inp))
-    (dynamic-require rep-start #f)))
+(define pre-readline-input-port
+  (let ([inp (current-input-port)] [outp (current-output-port)])
+    (and (eq? 'stdin (object-name inp))
+         (terminal-port? inp)
```

Comment:
Are these 2 conditions necessary? (Could the `let` on line 12 be replaced with the `dynamic-require`?)

If they are, then the docs should say something about them, e.g., "there two cases where readline can succeed but `pre-readline-input-port` is false"

Reply:
The documentation does say that it checks whether it's a terminal port (though it doesn't mention the check that it be named stdin), and says that it will not initialize in that case.  But it also says you can require readline/rep-start to unconditionally install the readline wrapper.

I would say the conditions are reasonable, also.  I'm not sure readline will run on an arbitrary port -- I bet it at least has to have a file descriptor backing it.

Reply:
After a quick look I stand corrected, it can work on eg. string ports.  But it is documented as checking that it is in a terminal unless you require readline/rep-start, and I'm fine with that.

Reply:
Ok. I think the doc for `pre-readline-input-port` should say that it checks for a terminal port named stdin. Maybe put this all in the contract: `(and/c input-port? terminal-port? (lambda (p) (eq? (object-name p) 'stdin))` ??